### ISSUE TITLE
Renovate no longer works with go_repository Bazel rules

### ISSUE THREAD

**Issue by rohansingh** [renovatebot/renovate#4402]
**What Renovate type are you using?**
Renovate Hosted

**Describe the bug**
As of a couple releases ago, the `go_repository` rule in Bazel looks like this example:

```python
go_repository(
    name = "com_github_pkg_errors",
    importpath = "github.com/pkg/errors",
    sum = "h1:iURUrRGxPUNPdy5/HRSm+Yj6okJ6UtLINN0Q9M4+h3I=",
    version = "v0.8.1",
)
```

Specifically, the new `sum` attribute has been added. Renovate is not able to handle these new `go_repository` rules.

**To Reproduce**

1.  Create repository with a Bazel `WORKSPACE` file and a `go_repository` rule like the one above.
2.  Configure Renovate to update Bazel dependencies.

**Expected behavior**
I expect a PR to be created to update the `go_repository` with a new version and sum value.

**Additional context**
This is how [the `go_repository` documentation](https://github.com/bazelbuild/bazel-gazelle/blob/master/repository.rst#go_repository) describes `sum`:

> A hash of the module contents. In module mode, `go_repository` will verify the downloaded module matches this sum. May only be set when `version` is also set.
>
> A value for `sum` may be found in the `go.sum` file or by running `go mod download -json <module>@<version>`.

**Comment by rarkins** [comment]
Isn't v0.8.1 the latest release though? Is there another package that has been updated recently that can be used as an example?

**Comment by rohansingh** [comment]
Yup, sorry, that was just an example of what the new field looks like. Here is an actual example of a `go_repository` rule I have in my `WORKSPACE` that is outdated:

```
go_repository(
    name = "org_golang_google_api",
    importpath = "google.golang.org/api",
    sum = "h1:VGGbLNyPF7dvYHhcUGYBBGCRDDK0RRJAI6KCvo0CL+E=",
    version = "v0.8.0",
)
```

The latest release is v0.9.0. However, no PR is created. Here's what I see in debug logs:

```
DEBUG: Checking bazel definition
{
  "def": "go_repository(\n    name = \"org_golang_google_api\",\n    importpath = \"google.golang.org/api\",\n    sum = \"h1:VGGbLNyPF7dvYHhcUGYBBGCRDDK0RRJAI6KCvo0CL+E=\",\n    version = \"v0.8.0\",\n)"
}
DEBUG: (org_golang_google_api)(dependency="org_golang_google_api")
INFO: Failed to find dependency in bazel WORKSPACE definition
{
  "def": "go_repository(\n    name = \"org_golang_google_api\",\n    importpath = \"google.golang.org/api\",\n    sum = \"h1:VGGbLNyPF7dvYHhcUGYBBGCRDDK0RRJAI6KCvo0CL+E=\",\n    version = \"v0.8.0\",\n)"
}
```

I mean, it makes sense to me— the `sum` field is new so the `go_repository` rule no longer looks like what Renovate expects. This changed with [Gazelle v0.18](https://github.com/bazelbuild/bazel-gazelle/releases/tag/0.18.0).
